hive/src/main/java/org/elasticsearch/hadoop/hive/pushdown/HiveTreeBuilder.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.hadoop.hive.pushdown;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hive.ql.plan.ExprNodeDesc;
+import org.apache.hadoop.hive.ql.plan.ExprNodeGenericFuncDesc;
+import org.elasticsearch.hadoop.hive.pushdown.node.ConstantNode;
+import org.elasticsearch.hadoop.hive.pushdown.node.FieldNode;
+import org.elasticsearch.hadoop.hive.pushdown.node.Node;
+import org.elasticsearch.hadoop.hive.pushdown.node.OpNode;
+import org.elasticsearch.hadoop.util.FieldAlias;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Build parse from Hive {@link ExprNodeDesc} to a tree
+ */
+public class HiveTreeBuilder {
+
+    protected static Log log = LogFactory.getLog(HiveTreeBuilder.class);
+
+    protected SargableParser sargableParser = null;
+
+    /**
+     * load es.mapping,names ,then get the field mmaping of hive to elasticsearch
+     */
+    protected FieldAlias fieldAlias = null;
+
+    public HiveTreeBuilder(FieldAlias fieldAlias, SargableParser sargableParser) {
+        this.sargableParser = sargableParser;
+        this.fieldAlias = fieldAlias;
+    }
+
+    /**
+     * get real elasticsearch field name.
+     *
+     * @param field
+     * @return when mapping contain filed, return the mapping value. otherwise return itself.
+     */
+    private String getMappingField(String field) {
+        if (fieldAlias == null) return field;
+        String mappingField = fieldAlias.toES(field);
+        if (mappingField == null)
+            return field;
+        else
+            return mappingField;
+    }
+
+    public OpNode build(ExprNodeDesc exprNodeDesc) {
+        OpNode root = OpNode.createRootNode();
+        _build(root, exprNodeDesc);
+        root.checkNeedScanAllTable(sargableParser);
+        root.checkIsAllOptimizable(sargableParser);
+
+        return root;
+    }
+
+    private void _build(Node nowParent, ExprNodeDesc hiveNode) {
```

Comment:
Nit: Standard java method naming please, I'd use something like `doBuild` if you want to differentiate from above `build` method.

Reply:
done. method name changed from _build to doBuild